src/ContentMessages.js

```diff
@@ -25,15 +25,19 @@ import sdk from './index';
 import { _t } from './languageHandler';
 import Modal from './Modal';
 import RoomViewStore from './stores/RoomViewStore';
-
 import encrypt from "browser-encrypt-attachment";
+// eslint-disable-next-line camelcase
+import png_chunks_extract from "png-chunks-extract";
 
 // Polyfill for Canvas.toBlob API using Canvas.toDataURL
 import "blueimp-canvas-to-blob";
 
 const MAX_WIDTH = 800;
 const MAX_HEIGHT = 600;
 
+// scraped out of a macOS hidpi (5660ppm) screenshot png
+//                  5669 px (x-axis)      , 5669 px (y-axis)      , per metre
+const PHYS_HIDPI = [0x00, 0x00, 0x16, 0x25, 0x00, 0x00, 0x16, 0x25, 0x01];
```

Comment:
Do we need to define the macOS header here? The spec defines pretty clear behaviour for how to handle the chunk

Reply:
there isn’t anything macOS specific here?

Reply:
I guess not macOS specific, but the resolution has to be an exact match due to line 134

Reply:
@turt2live what were you thinking of here? we're already parsing the png correctly into chunks, and then checking whether the `pHYs` chunk contains hidpi metrics of any flavour.

The only slight cheekiness is that it only applies this fix if the DPI is 5660 pixels-per-metre (which macOS uses for its screenshots) rather than trying to support arbitrary resolutions, but the reason for this is that absolute DPI values for screen use (as opposed to print use) are illdefined at the best of times.  After all, who's to say that 1024x768 should be viewed on a 15" or 17" monitor, etc?

We could refine this perhaps by assuming that anything higher than 96dpi (3780ppm) should be randomly halved in size when being viewed in Riot/Web, but i can easily see there may be some source of higher-DPI images out there where the human intention is to be viewed at 1-image-pixel = 1-screen-pixel rather than being halved, as is the intent for macOS screenshots.  So let's start like this, and then broaden it when we find other images getting mishandled, rather than overshoot.

Reply:
We do not check that the pHYs header has hipdi metrics of any flavour, only that they have the flavour we want (5669 pixels per metre). I don't think it's impossible for us to handle other sizes. and I do think we should even if other sources of hidpi images are uncommon. We definitely do not need to handle a unit of zero (unknown) though - that's just a nightmare.